aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+            <option selected="selected" value="5">Spain</option>
+        </select>
+        <br /><button type="submit">Register</button>
+        <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+
+Option Group
+^^^^^^^^^^^^^^^^^^^^^
+
+The HTML  `<optgroup> <https://www.w3.org/wiki/HTML/Elements/optgroup>`__ element is generated when the view model contains one or more `SelectListGroup  <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectListGroup/index.html>`__ objects.
+
+The ``CountryViewModelGroup`` groups the  ``SelectListItem`` elements into the "North America" and "Europe" groups:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModelGroup.cs
+  :language: c#
+  :lines: 6-56
+  :dedent: 3
+  :emphasize-lines: 8-9,17,23,35,41,47,29
+
+The two groups are shown below:
+
+.. image:: forms/_static/grp.png  
+
+The generated HTML:
+
+.. code-block:: HTML 
+  :emphasize-lines: 3-12
+
+    <form method="post" action="/Home/IndexGroup">
+        <select id="Country" name="Country">
+            <optgroup label="North America">
+                <option value="MEX">Mexico</option>
+                <option value="CAN">Canada</option>
+                <option value="US">USA</option>
+            </optgroup>
+            <optgroup label="Europe">
+                <option value="FR">France</option>
+                <option value="ES">Spain</option>
+                <option value="DE">Germany</option>
+            </optgroup>
+        </select>
+        <br /><button type="submit">Register</button>
+        <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+
+Multiple select
+^^^^^^^^^^^^^^^^^^^^^
+
+The Select Tag Helper  will automatically generate the `multiple = "multiple" <https://www.w3.org/TR/html-markup/select.html#select.attrs.multiple>`__  attribute if the property specified in the ``asp-for`` attribute is an ``IEnumerable``. For example, given the following model:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModelIEnumerable.cs
+  :language: c#
+  :emphasize-lines: 6
+
+With the following view:
+
+.. literalinclude::   forms/sample/final/Views/Home/IndexMultiSelect.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+  
+Generates the following HTML:
+
+.. code-block:: HTML  
+  :linenos:
+  :emphasize-lines: 3
+
+  <form method="post" action="/Home/IndexMultiSelect">
+      <select id="CountryCodes" 
+      multiple="multiple" 
+      name="CountryCodes"><option value="MX">Mexico</option>
+  <option value="CA">Canada</option>
+  <option value="US">USA</option>
+  <option value="FR">France</option>
+  <option value="ES">Spain</option>
+  <option value="DE">Germany</option>
+  </select> 
+      <br /><button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+No selection
```

Comment:
> it's a Select TH with an options attribute - right?

No, it's an Options Tag Helper. The text looks fine since the fact it's a separate tag helper is an implementation detail.